crates/ruff_dev/src/generate_json_schema.rs

```diff
@@ -20,7 +21,17 @@ pub(crate) struct Args {
 }
 
 pub(crate) fn main(args: &Args) -> Result<()> {
-    let schema = schema_for!(Options);
+    let mut schema = schema_for!(Options);
+    // Remove RUF014 since it's not present without the off-by-default feature `unreachable-code`
+    if let Some(Schema::Object(schema_object)) = &mut schema.definitions.get_mut("RuleSelector") {
+        schema_object
+            .enum_values
+            .as_mut()
+            .unwrap()
+            .retain(|field| field.as_str().unwrap() != "RUF014");
```

Comment:
We may also want to remove this from `mod clap_completion`, I think we did that in the past for these rules.